hibernate-core/src/main/java/org/hibernate/engine/spi/ActionQueue.java

```diff
@@ -1006,7 +1011,7 @@ private BeforeTransactionCompletionProcessQueue(SessionImplementor session) {
 		public void beforeTransactionCompletion() {
 			while ( !processes.isEmpty() ) {
 				try {
-					processes.poll().doBeforeTransactionCompletion( session );
+					NullnessUtil.castNonNull( processes.poll() ).doBeforeTransactionCompletion( session );
```

Comment:
I'm assuming that what you were aiming at is:
```
public interface Queue<E> extends Collection<E> {
    @EnsuresNonNullIf(expression="isEmpty()", result=true)
    boolean poll();
}
```
This one works, the well-intended suggestion tbh doesn't make sense to me, and also doesn't work ;)

Reply:
I think it was because the stub needed the poll() method also to be defined in there:
```
public interface Queue<E> extends Collection<E> {
    @Override
    @EnsuresNonNullIf(expression="poll()", result=false)
    @EnsuresNonNullIf(expression="peek()", result=false)
    boolean isEmpty();
    E poll();
    E peek();
}
```
Like this, it appears to work as expected ... who knew :shrug: